eval.c

```diff
@@ -1374,24 +1374,51 @@ sexp sexp_stream_portp_op (sexp ctx, sexp self, sexp_sint_t n, sexp port) {
 #endif
 
 #if SEXP_USE_STATIC_LIBS
-#if SEXP_USE_STATIC_LIBS_NO_INCLUDE
+#if SEXP_USE_STATIC_LIBS_EMPTY
+struct sexp_library_entry_t* sexp_static_libraries;
+#elif SEXP_USE_STATIC_LIBS_NO_INCLUDE
 extern struct sexp_library_entry_t* sexp_static_libraries;
 #else
 #include "clibs.c"
 #endif
+void sexp_add_static_libraries(struct sexp_library_entry_t* libraries)
+{
+  struct sexp_library_entry_t *entry, *table;
+
+  if (!sexp_static_libraries) {
+    sexp_static_libraries = libraries;
+    return;
+  }
+
+  for (table = sexp_static_libraries; ;
+       table = (struct sexp_library_entry_t*)entry->init) {
+    for (entry = &table[0]; entry->name; entry++);
+    if (!entry->init) {
```

Comment:
Could you fix the indentation here, and either use braces for both for's or neither?

Reply:
I assumed that by "indentation" you meant the 4 spaces inadvertently used inside the if block.  If you'd like me to fix something else, let me know.  As far as the for statements are concerned, note that I cannot use no braces for the first (as it contains two statements) and also that the second is empty; it merely advances the `entry` pointer to the end of the list.  Do you want me to add an empty pair of braces to it?

Reply:
Oh, actually I misread this, I thought the if was inside the for block.  It's better the be clear and put the semi-colon on a separate line:

```
    for (entry = &table[0]; entry->name; entry++)
       ;
```

Reply:
Done.